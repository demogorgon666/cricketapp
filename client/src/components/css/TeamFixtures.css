.team-fixtures-container {
  margin: 220px 0px 0px 20px;
}

.fixture-container {
  margin: 50px 180px 15px 150px;
}

.fixture-details {
  border: 1px solid #16a085;
  border-radius: 8px;
  display: flex;
}

.fixture-col-1 {
  margin: 10px 0px 10px 0px;
  border-right: 1px solid #16a085;
  padding: 0px 100px 0px 60px;
}

.fixture-col-2 {
  margin: 10px 0px 10px 0px;
  border-right: 1px solid #16a085;
  padding: 0px 100px 0px 100px;
}

.fixture-col-3 {
  margin: 10px 0px 10px 0px;
  padding: 0px 100px 0px 120px;
}

.fixture-info-type {
  font-size: 20px;
  color: #2c3e50;
}

.fixture-info-one {
  font-size: 20px;
  color: #2c3e50;
  margin-bottom: 0px;
}

.fixture-info-two {
  font-size: 12px;
  color: #707070;
}
#test {
  color: rebeccapurple;
}
#fixture-container {
  margin: 100px 0px 0px 0px;
}
#fix-team {
  color: #16a085;
}

#map {
  margin: 500px 0px 0px 0px;
}

#fixture-details {
  display: grid;
}

#fix-main-grid {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  padding: 0px 10px 10px 10px;
  margin: 200px 0px 0px 0px;
}
#fixgrid-item1 {
  color: black;
  padding: 0px 0px 0px 20px;
  border: 1px solid #16a085;
  background-color: white;
}
#fixture {
  margin: 100px 0px 0px 0px;
}
