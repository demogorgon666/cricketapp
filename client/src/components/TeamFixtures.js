import React, { Component } from "react";
import { connect } from "react-redux";
import "./css/TeamFixtures.css";
import { getFixtures } from "../actions/Teams";

export class TeamFixtures extends Component {
  componentDidMount() {
    this.props.getFixtures({ team_name: this.props.teams.team_name });
    console.log(this.props.teams.team_name);
  }
  render() {
    return (
      <div id="fixture-container">
        <div id="fixture">
          {/* <div className="centered">
            <div className="cards-new">
              <div
                style={{ borderRadius: "8px 0px 0px 8px" }}
                className="cardtest"
              >
                <p className="p-card">All</p>
              </div>
              <div className="cardt20">
                <p className="p-card">Test</p>
              </div>
              <div className="cardt20">
                <p className="p-card">ODI</p>
              </div>
              <div
                style={{ borderRadius: "0px 8px 8px 0px" }}
                className="cardt20"
              >
                <p className="p-card">T20</p>
              </div>
            </div>
          </div> */}

          <div id="fix-main-grid">
            {this.props.fixtures.map(fixtures => (
              <div id="fixgrid-item1">
                {/* //main grid container */}
                <div id="fixture-container">
                  {/* first grid item */}
                  <p id="fix-date">{fixtures.date}</p>
                  {/* second grid item */}
                  <p id="fix-teams">
                    {fixtures.team_one}vs
                    {fixtures.team_two}
                  </p>
                  <p id="fix-venue">{fixtures.venue}</p>
                  <p id="fix-time">{fixtures.time}</p>
                  {/* <img
                    // id="pImage"
                    // className="playerImage"
                    className="main_img"
                    src={`data:image/jpeg;base64,${fixtures.news_image}`}
                  ></img> */}
                </div>
              </div>
            ))}
          </div>
        </div>
      </div>
    );
  }
}

const mapStateTostate = state => ({
  fixtures: state.TeamsReducer.fixtures,
});

export default connect(
  mapStateTostate,
  { getFixtures }
)(TeamFixtures);
