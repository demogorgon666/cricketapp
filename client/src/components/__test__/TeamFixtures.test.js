import React from "react";
import { shallow } from "enzyme";
import { TeamFixtures } from "../TeamFixtures";
// import { getFixtures } from "../../actions/Teams";
const getFixtures = jest.fn();
// const fixtures = jest.fn();
const getPrediction = jest.fn();
const teams = [
  {
    team_name: "Australia",
  },
];

const fixtures = [
  {
    result: [{ date: "12-12-12", team_one: "India", team_two: "Australia" }],
    images: [{ team_image: "" }],
  },
];

const wrapper = shallow(
  <TeamFixtures
    getFixtures={getFixtures}
    getPrediction={getPrediction}
    fixtures={fixtures}
    history={[]}
    teams={teams}
    match={{ isExact: true, params: { path: "/", url: "/" } }}
  />
);
describe("Test TeamInfo Component", () => {
  it("render the component", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("User should be able to see the date of the upcoming match", () => {
    expect(wrapper.find("#fix-date").text()).toBe("12-12-12");
  });

  it("User should be able to see the name of both the team", () => {
    expect(wrapper.find("#fix-teams").text()).toBe("IndiavsAustralia");
  });
});
