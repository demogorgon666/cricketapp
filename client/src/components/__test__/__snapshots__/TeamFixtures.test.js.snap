// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TeamInfo Component render the component 1`] = `
<div>
  <div
    className="team-fixtures-container"
  >
    <div
      className="centered"
    >
      <div
        className="cards-new"
      >
        <div
          className="cardtest"
          style={
            Object {
              "borderRadius": "8px 0px 0px 8px",
            }
          }
        >
          <p
            className="p-card"
          >
            All
          </p>
        </div>
        <div
          className="cardt20"
        >
          <p
            className="p-card"
          >
            Test
          </p>
        </div>
        <div
          className="cardt20"
        >
          <p
            className="p-card"
          >
            ODI
          </p>
        </div>
        <div
          className="cardt20"
          style={
            Object {
              "borderRadius": "0px 8px 8px 0px",
            }
          }
        >
          <p
            className="p-card"
          >
            T20
          </p>
        </div>
      </div>
    </div>
    <div
      className="fixture-container"
    >
      <div
        className="fixture-details"
      >
        <div
          className="fixture-col-1"
        >
          <p
            className="fixture-info-type"
          >
            T20
          </p>
        </div>
        <div
          className="fixture-col-2"
        >
          <p
            className="fixture-info-one"
          >
            Wed 02 - Sun 06 Oct
          </p>
          <p
            className="fixture-info-two"
          >
            9:30 (IST)
          </p>
        </div>
        <div
          className="fixture-col-3"
        >
          <p
            className="fixture-info-one"
          >
            India vs Bangladesh
          </p>
          <p
            className="fixture-info-two"
          >
            JCA Stadium, Ranchi
          </p>
        </div>
      </div>
    </div>
    <div
      className="fixture-container"
    >
      <div
        className="fixture-details"
      >
        <div
          className="fixture-col-1"
        >
          <p
            className="fixture-info-type"
          >
            T20
          </p>
        </div>
        <div
          className="fixture-col-2"
        >
          <p
            className="fixture-info-one"
          >
            Wed 02 - Sun 06 Oct
          </p>
          <p
            className="fixture-info-two"
          >
            9:30 (IST)
          </p>
        </div>
        <div
          className="fixture-col-3"
        >
          <p
            className="fixture-info-one"
          >
            India vs South Africa
          </p>
          <p
            className="fixture-info-two"
          >
            JCA Stadium, Ranchi
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
